### PR TITLE
Fix background audio setting & mic-only device support

### LINKED ISSUE #11920: User cannot initiate Audio Self Recording if they don't have a webcam
## Bug Description

When someone wants to record an audio file on a computer that does not have a webcam, they get a message that says “Error: Could not initialize recording. Reason: No Camera Found.” [Seen on Windows 11, Chrome, on two machines that do not have web cameras (desktop, laptop w/o webcam)] - Reporter: Sam
Of note: When you do get this error, the buttons below that say “Record Video” and “Take a photo” look to still be enabled, but nothing happens when you click on them.

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

### LINKED ISSUE #11921: Media Recording: Audio is incorrectly inserted on story instead of page
## Bug Description

Inserting an Audio recording defaults to inserting that audio as the background audio for the entire story, which is different from what was expected.

## Expected Behaviour

The self recorded Audio should be inserted into the page only, not the entire story. Users should be able to self-record different audio for each story page.

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

### PR DESCRIPTION
**Pull request by merapi**
## Summary

This PR fixes two issues with Background Audio:
#11921
#11920

Mic-only support:
If you have a device without a built-in camera please test this PR. You can also attach the camera and check edge cases - I did a bunch of testing with or without resetting the permissions at different stages of the process.
The worst thing that can happen is you start a fresh session (after perms reset) and grant audio-only permission and then attach the camera - you will have to reload the page for the camera permission popup.

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. You need a device without a built-in camera.
2. Check if the audio recording works.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #11921
Fixes #11920

### CODE REVIEW

File: packages/story-editor/src/components/mediaRecording/provider.js [hunk 1]
```diff
@@ -143,12 +143,23 @@ function MediaRecordingProvider({ children }) {
     // granted permission, so it's not easy to detect.
     // TODO: Figure out how to retry without microphone if possible.
     mediaStreamConstraints: {
-      audio: audioInput && hasAudio ? { deviceId: audioInput } : true,
-      video: videoInput && hasVideo ? { deviceId: videoInput } : true,
+      audio: hasAudio ? (audioInput ? { deviceId: audioInput } : true) : false,
+      video: hasVideo ? (videoInput ? { deviceId: videoInput } : true) : false,
```
Comment (merapi) [inline packages/story-editor/src/components/mediaRecording/provider.js:147]: For the mic-only device we need to request a device with `video: false` - we do that when one of the errors occurs.

**Review by spacedmonkey — APPROVED** [review]
Seems to work well.

File: packages/story-editor/src/components/mediaRecording/footer.js [hunk 2]
```diff
@@ -318,12 +326,18 @@ function Footer({ captureImage, videoRef }) {
       resource: objectPick(resource, Object.keys(BackgroundAudioPropTypeShape)),
     };
 
+    const backgroundElement = currentPage?.elements.find(
+      ({ isBackground }) => isBackground
+    );
```
Comment (swissspidy) [inline packages/story-editor/src/components/mediaRecording/footer.js]: Let's move this to the `useStory` selector so that it already returns the `backgroundElementId` directly and not just the whole `currentPage` object

File: packages/story-editor/src/components/mediaRecording/provider.js [hunk 3]
```diff
@@ -143,12 +143,23 @@ function MediaRecordingProvider({ children }) {
...
     },
     onStop,
   });
 
+  useEffect(() => {
+    if (
+      error?.name === 'NotFoundError' ||
+      error?.name === 'NotReadableError' ||
+      error?.name === 'OverConstrainedError'
```
Comment (swissspidy) [inline packages/story-editor/src/components/mediaRecording/provider.js:156]: IIRC in some browsers this error can occur even when the user has not yet granted permission, so I hope this does not cause any issue in some edge cases. Just noting it here for visibility.

**Review by swissspidy — COMMENTED** [review]
Unfortunately I don't think I have a way to test this, but the change looks reasonable.

Just one nit

File: packages/story-editor/src/components/mediaRecording/footer.js (hunk 2 above)
Comment (merapi) [reply packages/story-editor/src/components/mediaRecording/footer.js]: https://github.com/GoogleForCreators/web-stories-wp/pull/11939/commits/5beefd2fd0cd2be00e0c81c05f23fdf2579f4a8a

**Comment by merapi** [comment]
This is what it looks like when you don't have a camera, only a mic (tested on W10 Edge/FF/Opera).
`NotFoundError` is thrown because it cannot find a device with `video/audio=true` then we set `hasVideo=false` so it requsts `video=false, audio=true` device and we display Audio Recording UI.

While the message is correct (No camera found) maybe we can add something about the audio-only option if they grant mic permissions?

image

File: packages/story-editor/src/components/mediaRecording/provider.js (hunk 3 above)
Comment (merapi) [reply packages/story-editor/src/components/mediaRecording/provider.js:156]: I haven't seen it anywhere, for me, it always looks like this:
https://github.com/GoogleForCreators/web-stories-wp/pull/11939#issuecomment-1188305242

File: packages/story-editor/src/components/mediaRecording/provider.js [hunk 4]
```diff
@@ -143,12 +143,23 @@ function MediaRecordingProvider({ children }) {
...
+    ) {
+      setHasVideo(false);
```
Comment (miina) [inline packages/story-editor/src/components/mediaRecording/provider.js:158]: Q: Should this influence the "Enable Video" quick-action as well? Can't test it unfortunately but looks like the button would still allow turning video on again (even though I suppose it'd give the error again and thus basically do nothing). Thoughts?

**Review by miina — APPROVED** [review]
LGTM! Also tried it out by setting the error in code directly, seems to work well like that.

File: packages/story-editor/src/components/mediaRecording/provider.js (hunk 4 above)
Comment (merapi) [reply packages/story-editor/src/components/mediaRecording/provider.js:158]: In the next line we do `setVideoInput(null);` which hides the enable/disable video icon.

**Comment by felipebochehin87** [comment]
Audio is added only to the page that is selected. If I clone this page, audio is cloned. If I add a new page, no audio is added to it. And if I have multiple pages and I record an audio, only the selected page has the audio linked to it.
